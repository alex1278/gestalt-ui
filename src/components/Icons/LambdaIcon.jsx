import React from 'react';
import PropTypes from 'prop-types';
import SVGIcon from '@material-ui/core/SvgIcon';

const LambdaIcon = ({ size, ...rest }) => (
  <SVGIcon viewBox="0 0 364.7 364.7" style={{ fontSize: `${size}px` }} {...rest}>
    <path d="M254.1,345.9l-63.1-159.8L98.6,331.9h-58l129.2-208.6l-14.5-41.2H105V32.9h86.1l98.8,251.4l43.5-14.2l14.5,48.7 L254.1,345.9z" />
  </SVGIcon>
);

LambdaIcon.propTypes = {
  size: PropTypes.number
};

LambdaIcon.defaultProps = {
  size: 24,
};

export default LambdaIcon;
