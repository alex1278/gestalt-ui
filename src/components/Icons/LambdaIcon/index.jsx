import React from 'react';
import Isvg from 'react-inlinesvg';
import styled from 'styled-components';
import iconSVG from 'assets/icons/lambda.svg';

const EnhancedIcon = styled(Isvg)`
  padding: .4em;
`;

const LambdaIcon = () => (
  <EnhancedIcon src={iconSVG} />
);

export default LambdaIcon;
