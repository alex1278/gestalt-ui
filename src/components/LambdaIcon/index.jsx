import React from 'react';
import Isvg from 'react-inlinesvg';
import styled from 'styled-components';
import iconSVG from '../../assets/icons/lambda_lc.svg';

const EnhancedIcon = styled(Isvg)`
  padding: .7em;
  fill: red;
`;

const LambdaIcon = () => (
  <EnhancedIcon src={iconSVG} />
);

export default LambdaIcon;
