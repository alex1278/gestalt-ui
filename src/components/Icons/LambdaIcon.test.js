import React from 'react';
import { render } from 'react-testing-library';
import LambdaIcon from './LambdaIcon';


describe('(Component) LambdaIcon', () => {
  it('renders component without exploding', () => {
    const { container } = render(<LambdaIcon />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
