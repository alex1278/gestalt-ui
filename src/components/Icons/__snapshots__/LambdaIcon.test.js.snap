// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) LambdaIcon renders component without exploding 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root-1"
  focusable="false"
  role="presentation"
  style="font-size: 22px;"
  viewBox="0 0 364.7 364.7"
>
  <path
    d="M254.1,345.9l-63.1-159.8L98.6,331.9h-58l129.2-208.6l-14.5-41.2H105V32.9h86.1l98.8,251.4l43.5-14.2l14.5,48.7 L254.1,345.9z"
  />
</svg>
`;
